src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,994 @@
+        for( auto b : range( region.nb_boundaries() ) )
+        {
+            const Surface3D& surface = region.boundary( b );
+            const Interface3D& interface = static_cast< const Interface3D& >(
+                surface.parent( Interface3D::type_name_static() ) );
+            if( GMGE::is_stratigraphic_limit( interface.geological_feature() ) )
+            {
+                horizons.insert( &interface );
+            }
+        }
+
+        if( horizons.size() >= 1 && horizons.size() <= 2 )
+        {
+            // TODO: the following detection is not working if the region
+            // doesn't contain a single face of one or even both horizons
+            std::vector< gmge_id > horizons_list;
+            for( auto h : horizons )
+            {
+                horizons_list.push_back( h->gmge() );
+            }
+            std::sort( horizons_list.begin(), horizons_list.end() );
+            for( auto& unit : unit_2_info_ )
+            {
+                std::vector< gmge_id > cur_horizons_list;
+                if( unit.second.interface_top_.is_defined() )
+                {
+                    cur_horizons_list.push_back( unit.second.interface_top_ );
+                }
+                if( unit.second.interface_base_.is_defined() )
+                {
+                    cur_horizons_list.push_back( unit.second.interface_base_ );
+                }
+                std::sort( cur_horizons_list.begin(), cur_horizons_list.end() );
+
+                if( cur_horizons_list.size() != horizons_list.size() )
+                {
+                    continue;
+                }
+                std::vector< gmge_id > result;
+                std::set_intersection( horizons_list.begin(),
+                    horizons_list.end(), cur_horizons_list.begin(),
+                    cur_horizons_list.end(), std::back_inserter( result ) );
+                if( result.size() == horizons_list.size() )
+                {
+                    auto id = interp_2_geo_entity_.find( unit.first );
+                    ringmesh_assert( id != interp_2_geo_entity_.end() );
+                    return id->second;
+                }
+            }
+        }
+        return gmge_id();
+    }
+
+    void GeoModelBuilderRESQMLImpl::read_rank_interpreation(
+        StratigraphicColumnRankInterpretation* rank_interp )
+    {
+        showAllMetadata( rank_interp );
+
+        for( auto unit : rank_interp->getStratigraphicUnitInterpretationSet() )
+        {
+            const gmge_id layer = builder_.geology.create_geological_entity(
+                Layer3D::type_name_static() );
+
+            builder_.geology.set_geological_entity_geol_feature(
+                layer, GMGE::GEOL_FEATURE::STRATI_UNIT );
+
+            interp_2_geo_entity_[unit] = layer;
+
+            unit_2_info_[unit].layer_ = layer;
+            if( unit->getInterpretedFeature() != nullptr )
+            {
+                const Layer3D& layer3d =
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "layer3d" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1481)